submissions/antonkottans/memory-pair-game/script.js

```diff
@@ -0,0 +1,206 @@
+document.addEventListener("DOMContentLoaded", () => {
+    let timerId = undefined;
+
+    const startTimer = () => {
+        document.querySelectorAll(".opened");
+        let minutes = 0;
+        let seconds = 0;
+        let timerOutput = document.querySelector(".timer");
+        const increaseTimer = () => {
+            seconds++;
+            if (seconds === 60) {
+                minutes++;
+                seconds = 0;
+            }
+        };
+        const getFormattedTimer = () => {
+            return `${minutes < 10 ? "0" : ""}${minutes}:${
+                seconds < 10 ? "0" : ""
+            }${seconds}`;
+        };
+        timerId = setInterval(() => {
+            increaseTimer();
+            if (minutes === 60) timerOutput.textContent = "you lost";
+            timerOutput.textContent = getFormattedTimer();
+        }, 1000);
+    };
+
+    const createClickHandler = () => {
+        let openned = 0;
+        let flipped = undefined;
+        return ({ target }) => {
+            if (
+                target.matches(".back") &&
+                target.parentNode.childElementCount === 2
+                && !target.closest('*[class*="card_"].flip-to-face')
+            ) {
+                target
+                    .closest('*[class*="card_"]')
+                    .classList.add("flip-to-face");
+
+                setTimeout(() => {
+                    let currentCardContainer =
+                        target.closest('*[class*="card_"]');
+                    let currentCardFace =
+                        target.closest('*[class*="card_"]').children[1];
+
+                    if (!flipped) {
+                        flipped = {
+                            container: currentCardContainer,
+                            face: currentCardFace,
+                        };
+                    } else {
+                        if (
+                            flipped.face.getAttribute("src").slice(9, -4) ===
```

Comment:
As I can see here, you're keeping information about the state of the game (which cards are open and which are closed) right in the DOM elements. This is wrong, because DOM elements are easily editable from the browser element inspector, so a user can cheat in your game, manually writing `opened` or other classes in these elements. 

You should find a way to store such information in JS and make DOM only _represent_ this information visually. DOM should __NEVER__ be the source of data except when you explicitly wait some information from a user, like in a login form. And you should __NEVER__ store internal information in DOM elements.

Hint: you could use an array of `true`/`false` values representing each card and when some card is flipped, get it's id and connect it to the number of `true`/`false` value in this array. You'll keep data about cards state in one place, not reachable to user. Also, win condition check writes itself this way - just check for the whole array to be `true`.